### PR TITLE
Fix oauth credfile permissions

### PR DESCRIPTION
**Pull request by dilipped**
Fixes https://github.com/GoogleCloudPlatform/gsutil/issues/994

Depends on https://github.com/gsutil-mirrors/oauth2client/pull/1

Since https://github.com/googleapis/oauth2client is now archived and in read-only mode, we have decided to move the library code to the `gslib/vendored` section.

### CODE REVIEW

File: gslib/wildcard_iterator.py [hunk 1]
```diff
@@ -238,8 +238,8 @@ def __iter__(self,
             if obj_or_prefix.datatype == CloudApi.CsObjectOrPrefixType.OBJECT:
               gcs_object = obj_or_prefix.data
               if prog.match(gcs_object.name):
-                if not suffix_wildcard or (StripOneSlash(gcs_object.name)
-                                           == suffix_wildcard):
+                if not suffix_wildcard or (StripOneSlash(
+                    gcs_object.name) == suffix_wildcard):
```
Comment (thomasmaclean) [inline gslib/wildcard_iterator.py:242]: Assuming this isn't doing anything you can just remove this.

File: gsutil.py [hunk 2]
```diff
@@ -74,7 +74,6 @@ def OutputAndExit(message):
     ('funcsigs', ''),  # mock dependency
     ('google-reauth-python', ''),  # Package name: google_reauth
     ('pyu2f', ''),  # google_reauth dependency
-    ('oauth2client', ''),
```
Comment (thomasmaclean) [inline gsutil.py:77]: Why did you move this out of `third_party` and move it into vendored? You could just change the location of the original submodule and keep everything else set up in the same way.

File: gslib/wildcard_iterator.py (hunk 1 above)
Comment (thomasmaclean) [inline gslib/wildcard_iterator.py]: Assuming this isn't doing anything you can just remove this.

Comment (dilipped) [reply gslib/wildcard_iterator.py]: Yep, these changes were made because I ran `python3 -m yapf -irp .`
But Travis CI failed anyway. Turns out, yapf behaves differently for python3.6 vs python3.7. Running in python3.6 removed this change so we should be good now.

File: gsutil.py (hunk 2 above)
Comment (dilipped) [reply gsutil.py:77]: This was done for PyPI. Because we are not maintaining our own version of oauth2client and not relying on the PyPI version, it was moved from here to `vendored_lib_dirs` in  `gslib.__init__.py` When gsutil gets installed via PyPI and invoked directly using `gslib.__main__.py` we might end up including the incorrect version.  The note on Line 91 explains this to some extent.

Comment (dilipped) [reply gsutil.py:77]: To give some context, it is the same reason why boto is not included here, and is present under the `vendored_lib_dirs` section.
